File: frontend/common/src/types/hobby.ts
export interface Hobby {
  id: number;
  name: string;
  category: string;
  startedAt: string | null;
  entries: HobbyProgressEntry[];
  createdAt: string;
}

export interface HobbyProgressEntry {
  id: number;
  hobbyId: number;
  entryDate: string;
  note: string | null;
  milestone: boolean;
  photoUrl: string | null;
  metadata: Record<string, unknown>;
  createdAt: string;
}

File: backend/newsletter-api/src/main/java/com/evalieu/newsletter/model/Hobby.java
package com.evalieu.newsletter.model;

import java.time.Instant;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "hobbies")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Hobby {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@Column(nullable = false)
	private String name;

	@Column(length = 100)
	private String category;

	@Column(name = "started_at")
	private LocalDate startedAt;

	@Column(name = "created_at", nullable = false)
	private Instant createdAt;

	@OneToMany(mappedBy = "hobby")
	@Builder.Default
	private List<HobbyProgressEntry> entries = new ArrayList<>();
}
